src/precice/impl/SolverInterfaceImpl.cpp

```diff
@@ -381,6 +381,8 @@ double SolverInterfaceImpl::advance(
   PRECICE_DEBUG("Handle exports");
   handleExports();
 
+  resetWrittenData();
```

Comment:
Is this issue severe enough to enforce this safe-guard on all our user?
Maybe it would be an idea to enable it only in Debug builds:

```suggestion
#ifndef NDEBUG
  resetWrittenData();
#endif
```

Reply:
We decided to leave it enabled all the time. Everything else is surprising for users and inconsistent for developers.